### PR TITLE
[FIX] spreadsheet: falsy pivot values

### PR DESCRIPTION
**Pull request by dhrp-odoo**
## Description:

Previously, when entering a pivot and applying a filter domain, the pivot formula (without domain field and domain value in function parameter) would return 'FALSE' if it did not match any records. The code now checks the value and returns an empty string if it's 'FALSE'.

Task - [3888401](https://www.odoo.com/web#id=3888401&cids=2&menu_id=4720&action=333&active_id=2328&model=project.task&view_type=form)

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by rrahir** [comment]
Hi :)
I think there might be a misunderstanding between the task description, which seems to specifically address the TOTAL fomulas and the PR description which seems to address every pivot formula. Could you check this?
Also, a test is required ;-)

**Comment by dhrp-odoo** [comment]
@rrahir, I have revised both the commit and PR message for clarity. Regarding the test, I've attempted it previously and again now, but I am receiving `0` instead of an empty string when pivot does not match any data. Functionally, it is working as expected. I am pushing a broken test. Could you provide insight if you are aware of any potential reasons for this issue? Thanks :)

File: addons/spreadsheet/static/src/pivot/pivot_model.js [hunk 1]
```diff
@@ -289,7 +289,15 @@ export class SpreadsheetPivotModel extends PivotModel {
         const { cols, rows } = this._getColsRowsValuesFromDomain(domain);
         const group = JSON.stringify([rows, cols]);
         const values = this.data.measurements[group];
-        return (values && values[0][measure]) ?? "";
+
+        if (
+            values &&
+            Object.prototype.hasOwnProperty.call(values[0], measure) &&
```
Comment (VincentSchippefilt) [inline addons/spreadsheet/static/src/pivot/pivot_model.js:295]: what is this ?

Comment (dhrp-odoo) [reply addons/spreadsheet/static/src/pivot/pivot_model.js:295]: It checks if `values[0]` has a property called `measure`.

Previously, I used `values[0].hasOwnProperty`, but encountered an ESLint error while playing around with tests: `Do not access Object.prototype method 'hasOwnProperty' from target object no-prototype-builtins`.

File: addons/spreadsheet/static/tests/pivots/model/pivot_plugin_test.js [hunk 2]
```diff
@@ -622,6 +622,43 @@ QUnit.module("spreadsheet > pivot plugin", {}, () => {
         assert.strictEqual(getEvaluatedCell(model, "A3").value, 0);
     });
 
+    QUnit.debug(
```
Comment (rrahir) [inline addons/spreadsheet/static/tests/pivots/model/pivot_plugin_test.js:625]: don't foret to remove the debug

**Review by rrahir — CHANGES_REQUESTED** [review]
I added a commit (so don't push force pbut rather pull the branch locally) which fixes the mocking of read_group, it did not follow the server behaviour.

**Review by aab-odoo — APPROVED** [review]
Commit message: [FIX] web: ...

File: addons/spreadsheet/static/src/pivot/pivot_model.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhrp-odoo) [reply addons/spreadsheet/static/src/pivot/pivot_model.js]: You are correct; this line is unnecessary. It was initially added while experimenting with tests and addressing an ESLint error. Thank you for the commit related to mockRPC 😊. I took the liberty of updating the commit message as suggested by AAB.

[the remark this answers is not in the document]
Comment (rrahir) [reply addons/spreadsheet/static/src/pivot/pivot_model.js]: thanks a lot! the fix seems good :)

**Comment by rrahir** [comment]
@robodoo rebase-ff

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**